File: AI/DotNet/MCPTest/DemoApp1/Tools/Sales.cs
using System.ComponentModel;
using DemoApp.Models;
using ModelContextProtocol.Server;

namespace DemoApp.Tools;

[McpServerToolType]
public class Sales
{
    private readonly Inventory model = new Inventory();

    [McpServerTool]
    [Description("Gets total price in purchase of an available item")]
    public double? QuotePrice([Description("Singularized name of item in lower-case")] string item, [Description("Number of items")] int quantity)
    {
        var info = model.Store.GetValueOrDefault(item);
        if(info != null)
        {
            double discount = quantity < 10 ? 0 : 0.05;
            return quantity * info.UnitPrice * (1 - discount);
        }
        return null;
    }

}

File: AI/Java/MCPTest/demo1/src/main/java/app/tools/Sales.java
package app.tools;

import app.models.Inventory;

public class Sales {
    
    private final Inventory model = new Inventory();

    public Double quotePrice(String item, int quantity) {
        var info = model.store().get(item);
        if(info != null) {
            double discount = quantity < 10 ? 0 : 0.05;
            return quantity * info.unitPrice() * (1 - discount);
        }
        return null;
    }
}
